### DISCUSSION TITLE
How to attach a Lua debugger to the redbean Lua interpreter?

### DISCUSSION THREAD

**Discussion by chrisokuda** [jart/cosmopolitan#510] [Q&A]
I'm looking to attach a VSCode debugger (I use "Lua Debug" extension, but I can figure something out from CLI commands) into my redbean application, but I'm not sure where to start. I can debug my code, other than redbean functions/modules, using standard Lua, but cannot debug the redbean-specific stuff. I'm looking to be able to at the very least run a file with `redbean -i filename.lua` and attach a debugger, but it would be cool if I could attach a debugger to a running redbean server, too :)

Thanks for any assistance!

**Comment by pkulchenko** [comment]
I've been working on getting Lua debugging to work using [ZeroBrane Studio IDE](https://studio.zerobrane.com/) I've been developing, but it will only help with the Lua-level debugging. I plan to have a plugin completed and a blog post with instructions up before the end of the month. Will update this ticket when it's available. There is unlikely to be a way to attach the debugging to the Lua code, but with a small code change you should be able to debug Lua files and hooks.

**Comment by chrisokuda** [comment]
> I've been working on getting Lua debugging to work using [ZeroBrane Studio IDE](https://studio.zerobrane.com/) I've been developing

The link isn't working for me, but I'm interested in looking at your project.

> There is unlikely to be a way to attach the debugging to the Lua code, but with a small code change you should be able to debug Lua files and hooks.

Yea, I found a few resources showing how I could run a debugger inside the interpreter at the start of my code and connect to it but so far I haven't tried it because I wanted to see if anyone had done it already or had a "magic forumula" or something like that :)

I think I'll try that method for now, as a simple cli debugger is sufficient and in the meanwhile someone here might comment on a method.

Thanks for the response [pkulchenko](https://github.com/pkulchenko)

**Reply by pkulchenko** [reply]
Thanks @chrisokuda; I fixed the link (it was missing the scheme) in both comments.

**Reply by chrisokuda** [reply]
The IDE looks really nice @pkulchenko! I will try it out a bit later on. I probably should have replied in a thread to your original comment reply, but I don't use github much sorry!

**Reply by pkulchenko** [reply]
Keep in mind that it doesn't support redbean debugging just yet, but as I wrote earlier, I plan to add a plugin to enable that.
